libs/template/config.go

```diff
@@ -0,0 +1,198 @@
+	schemaBytes, err := os.ReadFile(schemaPath)
+	if err != nil {
+		return nil, err
+	}
+	schema := &jsonschema.Schema{}
+	err = json.Unmarshal(schemaBytes, schema)
+	if err != nil {
+		return nil, err
+	}
+
+	// Return config
+	return &config{
+		ctx:    ctx,
+		schema: schema,
+		values: make(map[string]any, 0),
+	}, nil
+}
+
+// Reads json file at path and assigns values from the file
+func (c *config) assignValuesFromFile(path string) error {
+	// Read the config file
+	configFromFile := make(map[string]any, 0)
+	b, err := os.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(b, &configFromFile)
+	if err != nil {
+		return err
+	}
+
+	// Cast any integer properties, from float to integer. Required because
+	// the json unmarshaller treats all json numbers as floating point
+	for name, floatVal := range configFromFile {
+		property, ok := c.schema.Properties[name]
+		if !ok {
+			return fmt.Errorf("%s is not defined as an input parameter for the template", name)
+		}
+		if property.Type != jsonschema.IntegerType {
+			continue
+		}
+		v, err := toInteger(floatVal)
+		if err != nil {
+			return fmt.Errorf("failed to cast value %v of property %s from file %s to an integer: %w", floatVal, name, path, err)
+		}
+		configFromFile[name] = v
+	}
+
+	// Write configs from the file to the input map, not overwriting any existing
+	// configurations.
+	for name, val := range configFromFile {
+		if _, ok := c.values[name]; ok {
+			continue
+		}
+		c.values[name] = val
+	}
+	return nil
+}
+
+// Assigns default values from schema to input config map
+func (c *config) assignDefaultValues() error {
+	for name, property := range c.schema.Properties {
+		// Config already has a value assigned
+		if _, ok := c.values[name]; ok {
+			continue
+		}
+
+		// No default value defined for the property
+		if property.Default == nil {
+			continue
+		}
+
+		// Assign default value if property is not an integer
+		if property.Type != jsonschema.IntegerType {
+			c.values[name] = property.Default
+			continue
+		}
+
+		// Cast default value to int before assigning to an integer configuration.
+		// Required because untyped field Default will read all numbers as floats
+		// during unmarshalling
+		v, err := toInteger(property.Default)
+		if err != nil {
+			return fmt.Errorf("failed to cast default value %v of property %s to an integer: %w", property.Default, name, err)
+		}
+		c.values[name] = v
+	}
+	return nil
+}
+
+// Prompts user for values for properties that do not have a value set yet
+func (c *config) promptForValues() error {
+	for name, property := range c.schema.Properties {
+		// Config already has a value assigned
+		if _, ok := c.values[name]; ok {
+			continue
+		}
+
+		// Initialize Prompt dialog
+		var err error
+		prompt := cmdio.Prompt(c.ctx)
+		prompt.Label = property.Description
+		prompt.AllowEdit = true
+
+		// Compute default value to display by converting it to a string
+		if property.Default != nil {
```

Comment:
Do we need to prompt if there is a Default value but it's not set? Shall we just assign default value?

Reply:
We would like to prompt the user, if possible. Consider for example a field called cloud_type. The default could be azure because a lot of our customers are on azure, but this is definitely something we would like to prompt for.